mmv1/templates/terraform/pre_create/network_security_security_profile_operation.go.erb

```diff
@@ -0,0 +1 @@
+project = ""
```

Comment:
I'm unfamiliar with this paradigm, so just checking, do you know if it is needed here?

Reply:
Centralizing the discussion here, given I was using this in multiple resources "empirically" and I got the same question - cc @melinath 

FYI when I say empirically, I copied this from the `AuthorizationPolicy` resource, part of `networkservice` as well, being a very similar resource, after seeing my resource not working. I then decided to understand better what I was doing. Here is the result of my investigation:

Without doing any modification, you would see all tests failing with the following error

```
Error: ../../../go/pkg/mod/github.com/modular-magician/terraform-provider-google-beta@v1.20.1-0.20240112173651-8e753b30b478/google-beta/services/networksecurity/resource_network_security_security_profile.go:425:4: not enough arguments in call to NetworkSecurityOperationWaitTime
    have (*"github.com/hashicorp/terraform-provider-google-beta/google-beta/transport".Config, map[string]interface{}, string, string, time.Duration)
    want (*"github.com/hashicorp/terraform-provider-google-beta/google-beta/transport".Config, map[string]interface{}, string, string, string, time.Duration)
```

TL;DR the operationWaitTime call seems different from other networksecurity calls and takes another param in input: the project. It doesn't need values. It can just be an empty string

Indeed, I checked the go file that was produced and this is what the NetworkSecurityOperationWaitTime function (prior to modifications looks like):

```
err = NetworkSecurityOperationWaitTime(
        config, res, "Updating SecurityProfile", userAgent,
        d.Timeout(schema.TimeoutUpdate))
```

`include_project: true` in the `SecurityProfile.yaml` file creates a project variable in the resulting built file and passes it into the `NetworkSecurityOperationWaitTime` function.

Indeed, after adding it and going again to the resulting go file:

- at the top of the create function, there's a declaration `var project string`

- the NetworkSecurityOperationWaitTime function becomes:
```
err = NetworkSecurityOperationWaitTime(
        config, res, project, "Creating SecurityProfile", userAgent,
        d.Timeout(schema.TimeoutCreate))
```

I honestly don't understand instead the reason for the custom_code block. They simply initialize that variable to an empty string, which I guess happens by default in go as soon as we declare it as a string (effect of `include_project: true`). Indeed, removing the custom code directive and the related pre* files has no negative effects on the tests.

If that's the case, I'll send another PR to remove them as well for the `AuthorizationPolicy` resource.

Please, let me know if this helps to clarify.

Reply:
If removing the custom code doesn't have any negative impacts, then it would be great to remove. Additionally, `include_project` does seem to actually be used in our templates but I'm not familiar with the 

include_project does seem to actually have an impact, but it is a very unusual flag - I only see 8 uses across the entire provider at the moment. According to https://github.com/GoogleCloudPlatform/magic-modules/blob/2031a558cd048257698c1b1297089551d69bad5a/mmv1/api/async.rb#L69, it's intended for resources that calculate the project from a selflink field (which doesn't seem to apply here). I'd say that it would be good to remove the usage here (and from the [PR I'm reviewing](https://github.com/GoogleCloudPlatform/magic-modules/pull/9802)) if possible.

Reply:
Taking a closer look at this, I think that `include_project: true` actually probably is a good thing to use here.

Operations are a shared class within a given service, which can cause some weird edge cases because we're not currently consistent about their _configuration_ being shared - some templates reference the product.yaml async block, and some reference resource async blocks. In particular, resource files (which call the operation) check whether to include a project based on the resource's config - but the operation go file for a service uses the async block from the first resource it finds in the service with `autogen_async: true` set.

So, because the first matching resource in the networksecurity product happens to generate an operation that requires the project argument, all resources in the product need to be sending that argument.

The project is ultimately just passed through to [`transport_tpg.SendRequest`](https://github.com/hashicorp/terraform-provider-google-beta/blob/73e5a29f0b8e69427045837c786d4096f5829e06/google-beta/transport/transport.go#L33), which uses it to set the X-Goog-User-Project header (for project billing overrides).

I don't think it makes sense to remove support for this from resources that previously supported it, so using `include_project: true` seems like a reasonable step.

Technically there's probably some additional work here to add support for extracting the project from the parent field (if that would ever be a project.) But I don't _think_ that's necessarily a blocker (especially if the parent would only ever be an organization or folder) - I'll check on this though.

Reply:
Thanks @melinath ! This definitely helped me to understand one more piece of mmv1 :)

Reply:
Just circling back on this, it looks like we've removed the custom code blocks, and now just use `include_project: true`, which I believe is the best option moving forward. I would advocate for unblocking this PR since it is matching an existing pattern, and if we would like to make improvements to this usage pattern, do so afterward across all impacted resources.

I'll wait to hear back from @melinath before merging anything.

Reply:
Caught up offline: sounds like we agree this is not a blocker, so we can move forward with merging.